src/main/java/com/coveo/saml/SamlClient.java

```diff
@@ -453,14 +469,18 @@ private void validateAssertion(Response response) throws SamlException {
   private void enforceConditions(Conditions conditions) throws SamlException {
     DateTime now = this.now != null ? this.now : DateTime.now();
 
-    if (now.isBefore(conditions.getNotBefore())) {
+    DateTime notBefore = conditions.getNotBefore();
+    DateTime skewedNotBefore = notBefore.minus(notBeforeSkew);
+    if (now.isBefore(skewedNotBefore)) {
       throw new SamlException(
-          "The assertion cannot be used before " + conditions.getNotBefore().toString());
+          "The assertion cannot be used before " + skewedNotBefore.toString() + " (" + notBefore.toString() +
```

Comment:
I'm not quite sure we want to include the allowable skew in the exception message. If that ends up being displayed to a caller, as exception messages often are, are we disclosing some information that we'd rather shouldn't?

I don't have super strong feelings on this, but when security is concerned tend to err on the side of caution. So maybe just keep the exception message as it was even if we do allow some flexibility about the date?

Reply:
That sounds fair enough to me. I reverted the error message.